operator/v1/types_ingress.go

```diff
@@ -282,6 +289,12 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the schema for securing an IngressController.
+type SecuritySpec struct {
+	// profile defines the schema for a security profile.
+	Profile configv1.SecurityProfileSpec `json:"profile"`
```

Comment:
What is the purpose of `SecuritySpec`?  I have a vague recollection that there could be fields we might want to add in the future that would not belong in the security profile, but I cannot recall specific examples, and I am wondering whether `SecuritySpec` should go in `configv1` along with `SecurityProfileSpec` (or maybe `SecuritySpec` isn't even needed at all).

Reply:
@Miciah as you mention, the intention of `SecuritySpec` is to support future `IngressController` security management capabilities. Currently, the focus is on managing secure network connections (i.e. TLS versions). I expect future security requirements outside of securing network connections to arise. As I mentioned in https://github.com/openshift/api/pull/350#issuecomment-509428587, I have a 2nd implementation where `SecurityProfile` provides a higher level of abstraction in `config.openshift.io`. In that case, I can see removing `SecuritySpec` from the `IngressController` api. @ironcladlou how would you like to proceed?

Reply:
Is this the suggestion?

```go
type IngressControllerSpec {
  // ...
  SecurityProfile configv1.SecurityProfileSpec `json:"securityProfile"`
}
```

Reply:
I wouldn't call it a suggestion, but yeah, I'm asking why we don't do what you describe.  I think there is a reason not to, but I'm interested in specific reasons.  I would like to verify (1) that those reasons exist and (2) that those reasons apply only to the ingress API (and do not belong in the config API).

Reply:
#### Option 1

```go
type IngressControllerSpec {
  // ...
  SecurityProfile configv1.SecurityProfileSpec `json:"securityProfile"`
}
```

Looks like:

```yaml
spec:
  securityProfile:
    type: Modern
```

A theoretical future ingress-specific security configuration would look like:

```yaml
spec:
  securityProfile:
    type: Modern
  otherSecurityFeature:
    field: value
```

#### Option 2

```go
type IngressControllerSpec {
  // ...
  Security IngressControllerSecuritySpec `json:"security"`
}

type IngressControllerSecuritySpec {
  Profile configv1.SecurityProfileSpec `json:"profile"`
}
```

Looks like:

```yaml
spec:
  security:
    profile:
      type: Modern
```

A theoretical future ingress-specific security configuration would look like:

```yaml
spec:
  security:
    profile:
      type: Modern
    otherSecurityFeature:
      field: value
```

One substantive difference is that in Option 2 all security related features (whether cluster-scoped/generic or specific to ingress) are nested under `.spec.security`, while additional security related fields in Option 1 become siblings under `.spec`.

Are there other differences? To me, the one I have identified seems purely cosmetic.